e2e_playwright/st_popover_test.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from playwright.sync_api import Page, expect
+
+from e2e_playwright.conftest import ImageCompareFunction, wait_for_app_run
+
+
+def test_popover_button_rendering(
+    themed_app: Page, assert_snapshot: ImageCompareFunction
+):
+    """Test that the popover buttons are correctly rendered via screenshot matching."""
+    popover_elements = themed_app.get_by_test_id("stPopover")
+    expect(popover_elements).to_have_count(5)
+
+    assert_snapshot(popover_elements.nth(0), name="st_popover-sidebar")
+    assert_snapshot(popover_elements.nth(1), name="st_popover-empty")
+    assert_snapshot(popover_elements.nth(2), name="st_popover-use_container_width")
+    assert_snapshot(popover_elements.nth(3), name="st_popover-normal")
```

Comment:
Does it make sense to add a screenshot for the dataframe popover?

Reply:
Yeah, I try to prevent the pollution of tests with UI aspects from unrelated elements. E.g, if I change a small aspect with dataframe, it would be very cumbersome to update many other unrelated tests (snapshots). And the dataframe frontend component has the tendency to slightly change with new glide-data-grid updates.